extensions/replication/ReplicationQueuePopulator.js

```diff
@@ -47,6 +44,10 @@ class ReplicationQueuePopulator extends QueuePopulatorExtension {
         if (sanityCheckRes) {
             return;
         }
+        // ALlow a non-versioned object if being replicated from an NFS bucket.
+        if (!isMasterKey(entry.key) && !queueEntry.getReplicationIsNFS()) {
```

Comment:
I think the logic should be `if (isMasterKey(entry.key) && !queueEntry.getReplicationIsNFS()) return;`

Reply:
Good catch!